Objects/moduleobject.c

```diff
@@ -823,11 +823,7 @@ module_dir(PyObject *self, PyObject *args)
             }
         }
         else {
-            const char *name = PyModule_GetName(self);
-            if (name)
-                PyErr_Format(PyExc_TypeError,
-                             "%.200s.__dict__ is not a dictionary",
-                             name);
+            PyErr_Format(PyExc_TypeError, "<module>.__dict__ is not a dictionary");
```

Comment:
Why not using %R to use repr(module)?

Reply:
I didn't think of it.  Would that produce a better result, if we had one of these broken, nameless, dict-less module objects?